beaker_kernel/lib/context.py

```diff
@@ -41,13 +46,25 @@ def __init__(self, beaker_kernel: "LLMKernel", agent_cls: "BaseAgent", config: D
         self.jinja_env = None
         self.templates = {}
         self.beaker_kernel = beaker_kernel
+        self.config = config
+        self.subkernel = self.get_subkernel()
+        subkernel_tools = collect_tools_from_object(self.subkernel)
         self.agent = agent_cls(
             context=self,
-            tools=[],
+            tools=subkernel_tools,
         )
-        self.config = config
-        self.subkernel = self.get_subkernel()
 
+        # Remove tools
+        # NOTE: Identical toolnames don't work
+        enabled_tools = [tool.split(".")[-1] for tool in self.agent.tools.keys()]
+        toggles = [attr[len(TOOL_TOGGLE_PREFIX):].lower() for attr in dir(self) if attr.startswith(TOOL_TOGGLE_PREFIX)]
```

Comment:
Since we're pinned to 3.10, we can use [str.removeprefix()](https://docs.python.org/3/library/stdtypes.html#str.removeprefix) instead of subsetting via indexing.